File: src/main/java/io/github/emcw/map/entities/Capital.java
package io.github.emcw.map.entities;

import com.google.gson.JsonObject;
import io.github.emcw.interfaces.ISerializable;
import lombok.Getter;

import static io.github.emcw.utils.GsonUtil.keyAsInt;
import static io.github.emcw.utils.GsonUtil.keyAsStr;

public class Capital implements ISerializable {
    @Getter final String name;
    @Getter final Location location;

    public Capital(JsonObject obj) {
        name = keyAsStr(obj, "name");
        location = new Location(keyAsInt(obj, "x"), keyAsInt(obj, "z"));
    }
}

File: src/main/kotlin/io/github/emcw/map/entities/Capital.kt
package io.github.emcw.map.entities

import com.google.gson.JsonObject
import io.github.emcw.interfaces.ISerializable
import io.github.emcw.utils.GsonUtil
import lombok.Getter

class Capital(obj: JsonObject?) : ISerializable {
    @Getter val name: String?
    @Getter val location: Location

    init {
        name = GsonUtil.keyAsStr(obj, "name")
        location = Location(GsonUtil.keyAsInt(obj, "x"), GsonUtil.keyAsInt(obj, "z"))
    }
}